models/kernels/evapotranspiration/EtWrapperFunction.hpp

```diff
@@ -0,0 +1,289 @@
+  //---------------------------------------------------------------------------------------------------------------
+  surf_rad_forcing.surface_skin_temperature_C = surface_rad_forcing->surface_skin_temperature_C;  // TODO from soil thermal model or vegetation model.
+
+  if(et_options.shortwave_radiation_provided=FALSE)
+  {
+    // populate the elements of the structures needed to calculate shortwave (solar) radiation, and calculate it
+    // ### OPTIONS ###
+    solar_options.cloud_base_height_known=FALSE;  // set to TRUE if the solar_forcing.cloud_base_height_m is known.
+
+    // ### PARAMS ###
+    solar_params.latitude_degrees  = solar_rad_params->latitude_degrees;   // THESE VALUES ARE FOR THE UNIT TEST
+    solar_params.longitude_degrees = solar_rad_params->longitude_degrees;  // THESE VALUES ARE FOR THE UNIT TEST
+    solar_params.site_elevation_m  = solar_rad_params->site_elevation_m;   // THESE VALUES ARE FOR THE UNIT TEST  
+
+    // ### FORCING ###
+    solar_forcing.cloud_cover_fraction         = solar_rad_forcing->cloud_cover_fraction; // THESE VALUES ARE FOR THE UNIT TEST 
+    solar_forcing.atmospheric_turbidity_factor = solar_rad_forcing->atmospheric_turbidity_factor; // 2.0 = clear mountain air, 5.0= smoggy air
+    solar_forcing.day_of_year                  = solar_rad_forcing->day_of_year; // THESE VALUES ARE FOR THE UNIT TEST
+    solar_forcing.zulu_time_h                  = solar_rad_forcing->zulu_time_h; // THESE VALUES ARE FOR THE UNIT TEST
+
+    calculate_solar_radiation(&solar_options, &solar_params, &solar_forcing, &solar_results);
+    surf_rad_forcing.incoming_shortwave_radiation_W_per_sq_m = 
+            solar_results.solar_radiation_horizontal_cloudy_flux_W_per_sq_m;
+  }
+  
+  // we must calculate the net radiation before calling the ET subroutine.
+  if(et_options.use_aerodynamic_method==FALSE) 
+  {
+    // NOTE don't call this function use_aerodynamic_method option is TRUE
+    et_forcing.net_radiation_W_per_sq_m=calculate_net_radiation_W_per_sq_m(&et_options,&surf_rad_params, 
+                                                                         &surf_rad_forcing);
+  }
+
+  if(et_options.use_energy_balance_method ==TRUE)
+    et_m_per_s=evapotranspiration_energy_balance_method(&et_options,&et_params,&et_forcing);
+  if(et_options.use_aerodynamic_method ==TRUE)
+    et_m_per_s=evapotranspiration_aerodynamic_method(&et_options,&et_params,&et_forcing,&inter_vars);
+  if(et_options.use_combination_method ==TRUE)
+    et_m_per_s=evapotranspiration_combination_method(&et_options,&et_params,&et_forcing,&inter_vars);
+  if(et_options.use_priestley_taylor_method ==TRUE)
+    et_m_per_s=evapotranspiration_priestley_taylor_method(&et_options,&et_params,&et_forcing,&inter_vars);
+  if(et_options.use_penman_monteith_method ==TRUE)
+    et_m_per_s=evapotranspiration_penman_monteith_method(&et_options,&et_params,&et_forcing,&inter_vars);
+
```

Comment:
I see a fairly consistent interface to each of these et methods...`(&et_options, &et_params, &et_forcing)`. The only difference is the` inter_vars`, is it possible to let these functions create and hold their own `internal_variables` structs? Also, this wrapper does a small bit of manipulation, but mostly just copies structure values.  What is the intent of this wrapper?  Can callers just call directly `evapotranspiration_energy_balance_method` or `evapotranspiration_penman_monteith_method`?